test/unit/db/trace.test.js

```diff
@@ -5,13 +5,13 @@
 
 'use strict'
 
-const tap = require('tap')
+const test = require('node:test')
+const assert = require('node:assert')
 const helper = require('../../lib/agent_helper')
 
-tap.test('SQL trace attributes', function (t) {
-  t.autoend()
+test('SQL trace attributes', async (t) => {
   t.beforeEach(function (t) {
-    t.context.agent = helper.loadMockedAgent({
+    t.mock.agent = helper.loadMockedAgent({
```

Comment:
Tap provided a specific `context` namespace on the test context (`t`) for us to store test scoped data. Unfortunately, `node:test` only provides a test scoped context. So we need to add our own namespace. I have been writing tests like:

```js
test('some feature', async t => {
	t.beforeEach(ctx => {
		ctx.nr = {}
		ctx.nr.agent = {agent: 'instance'}
	})

	t.afterEach(ctx => {
		unload(ctx.nr.agent)
	})

	await t.test('one'), (t, end) => {
		const { agent } = t.nr
		agent.someMethod((error) => {
			assert.equal(error, undefined)
			end()
		})
	})
})
```

What you have written in this test is adding to the test's scoped `mock` tracker entity (see https://nodejs.org/api/test.html#class-mocktracker). Which means we would be adding things to a namespace we don't control and could possibly encounter naming collisions.

In short, instead of attaching things to `t.mock`, please attach them to `t.nr`.